File: ABC/abc211/src/c.rs
use proconio::input;
use proconio::marker::Chars;

fn main() {
    input! {
        s: Chars,
    }

    const MOD: usize = 1_000_000_007;

    let t = "chokudai".chars().collect::<Vec<char>>();
    let m = t.len();
    let n = s.len();
    let mut dp = vec![vec![0; n + 1]; m + 1];

    for i in 0..=n {
        dp[0][i] = 1;
    }

    for i in 1..=m {
        for j in 1..=n {
            if t[i - 1] == s[j - 1] {
                dp[i][j] = (dp[i][j - 1] + dp[i - 1][j - 1]) % MOD;
            } else {
                dp[i][j] = dp[i][j - 1];
            }
        }
    }

    println!("{}", dp[8][n])
}

File: abc/abc211/c.py
MOD = 10 ** 9 + 7
S = input()
T = "chokudai"
N = len(S)
M = len(T)

# dp[i][j] := Sのj文字目までを"chokudai"のi文字目までで何通り作れるか
dp = [[0] * (N + 1) for _ in range(M + 1)]
for i in range(N + 1):
    dp[0][i] = 1

for i in range(1, M + 1):
    for j in range(1, N + 1):
        if T[i - 1] == S[j - 1]:
            dp[i][j] = (dp[i][j - 1] + dp[i - 1][j - 1]) % MOD
        else:
            dp[i][j] = dp[i][j - 1]

print(dp[M][N])
        
